src/cdn_definitions/schema.yaml

```diff
@@ -11,7 +11,211 @@ $id: http://release-engineering.github.io/cdn-definitions/schema.json
+
+  arch_list:
+    type: array
+    items:
+      $ref: "#/definitions/arch"
+
+  release_version:
+    type: string
+    pattern: "^[0-9]+\\.[0-9]+$"
+
+  major_version:
+    type: string
+    pattern: "^[0-9]+$"
+
+  signing_key:
+    type: string
+    pattern: "^(37036783|A5787476|FD431D51|F21541EB|37017186|897DA07A|DB42A60E|\\[\\])$"
+
+  product_id:
+    type: string
+    pattern: "^[0-9]+$"
+
+  product_id_list:
+    type: array
+    items:
+      $ref: "#/definitions/product_id"
+
+  signing_key_list:
+    type: array
+    items:
+      $ref: "#/definitions/signing_key"
+
+  release_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/release_version"
+
+  major_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/major_version"
+
+  major_or_release_version:
+    type: string
+    pattern: "^[0-9]+(\\.[0-9]+)*$"
+
+  # various product definition mappings
+  arch_to_version_list_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/arch"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/major_version_list"
+    additionalProperties: false
+
+  major_to_release_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  release_stream_mapping:
+    type: object
+    properties:
+      aus:
+        $ref: "#/definitions/release_version_list"
+      eus:
+        $ref: "#/definitions/release_version_list"
+      tus:
+        $ref: "#/definitions/release_version_list"
+      e4s:
+        $ref: "#/definitions/release_version_list"
+    requiredProperties:
+    - aus
+    - eus
+    - tus
+    - e4s
+    additionalProperties: false
+
+  current_latest_rhel_versions:
+    type: object
+    propertyNames:
+      type: string
+      pattern: "^(current|latest)\\_rhel[0-9]+\\_(beta|dist)$"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  env_to_releasever_mapping:
+    type: object
+    properties:
+      qa:
+        $ref: "#/definitions/current_latest_rhel_versions"
+      stage:
+        $ref: "#/definitions/current_latest_rhel_versions"
+      prod:
+        $ref: "#/definitions/current_latest_rhel_versions"
+    required:
+    - qa
+    - stage
+    - prod
+    additionalProperties: false
+
+  product_cert_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  # TPS stream configuration
+  rhel_ll_tps_stream_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/release_version"
+    patternProperties:
+      ".*":
+        type: string
+        pattern: "^RHEL\\-\\%s\\-(LL|AUS)\\-\\%s$"
```

Comment:
I'm not sure the restriction here in the schema is right.  Looking at the actual data, it's always used the AUS style in RHEL6 and later.

I would expect the future is one of:

- Either: we *always* stick with the AUS style from here onwards.
   - In that case, this isn't too useful, the cdn-utils code may as well be "if RHEL5, then use LL format, otherwise use AUS format"
- Or: there is some case later on to introduce a new style
   - In that case, the schema as implemented now can't support it since it's restricted to only two possibilities, so you would have to go and update the schema to support a new value, then go and update your data to use exactly that value.

I'm guessing the former is more likely and I'd rather just not have this in the data.  I'm looking at `RHEL_LL_TPS_STREAM_MAPPING` and to me it seems that all the time spent updating this field across many release versions has been wasted, what with the value always being the same since RHEL6.  Some things would be better off *not* configurable if we want to encourage consistency going forward.